FaunaDB.Client/Types/Decoder.cs

```diff
@@ -143,22 +145,66 @@ static object DecodeIntern(Value value, Type dstType)
 
                 case TypeCode.Object:
                     if (typeof(DateTimeOffset) == dstType)
-                        return new DateTimeOffset(ToDateTime(value));
+                        if (forceType == typeof(StringV) && value.GetType() == typeof(StringV))
+                            if (value != null && value != NullV.Instance)
+                                return new DateTimeOffset(Convert.ToDateTime(ToString(value)));
+                            else
+                                return null;
+                        else
+                            return new DateTimeOffset(ToDateTime(value));
 
                     if (typeof(byte[]) == dstType && value is BytesV)
-                        return ToByteArray(value);
+                        if (forceType == typeof(StringV) && value.GetType() == typeof(StringV))
+                            if (value != null && value != NullV.Instance)
+                                return JsonConvert.DeserializeObject<byte[]>(ToString(value));
```

Comment:
I'm still not convinced this is the correct way of doing this, in my opinion users could accomplish the same thing doing:
```csharp
        class Address
        { }

        class User
        {
            [FaunaField("addresses")]
            public string AddressesStr
            {
                get
                {
                    return JsonConvert.SerializeObject(Addresses);
                }

                set
                {
                    Addresses = JsonConvert.DeserializeObject<List<Address>>(value);
                }
            }

            [FaunaIgnore]
            public List<Address> Addresses { get; set; }
        }
```
this is more verbose but it will keep the driver simpler and give more flexibility to the user.

Reply:
My view is the whole point of a driver is to make the code that uses it less verbose and much easier to use. Ive been doing that exact thing to get around for Uri types. I mean for me the Collections are a non issue but someone in the future might find it useful. Your previous suggestions have been resolved in 052582c

Reply:
Ok. In a second look, I'm in favor of Marrony's comment as well. Maybe we could remove that specific serialisation use cases from the Encoder/Decoder and add some sort of utility helper for the user. 

Lets try to figure it out a simpler "interface" to enable users to (de)serialize those types.

Reply:
removed that